### PR TITLE
Asteroid coating

### PR DESCRIPTION
**Pull request by Deepsloth**

### CODE REVIEW

**Comment by Deepsloth** [comment]
Description: http://www.freeorion.org/forum/viewtopic.php?f=15&t=9746

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -6373,6 +6373,24 @@ PHILOSOPHER_SPECIAL_DESC
...
 
+ASTEROID_COATING_SPECIAL
+Asteroid Coating
+
+ASTEROID_COATING_SPECIAL_DESC
+'''An outpost or colony on these asteroids will grand a structure bonus of +10% to all ships with Asteroid hulls in the empire.
+
+These asteroids are infected by transdimensional worms that eat tunnels through the rock, leaving behind a sticky slime that seems to automatically spread over the whole surface of the affected asteroid and prevents it from breaking appart.
+Applied to ships build with asteroid hulls the slime will naturally increase their durability. While scientists were unable to reproduce the substance or catch a living worm, they found out that just a small amount of fresh slime is enough to enhance whole fleets.'''
```
Comment (MatGB) [inline default/stringtables/en.txt]: *built

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -6373,6 +6373,24 @@ PHILOSOPHER_SPECIAL_DESC
...
+
+ASTEROID_COATING_OWNED_SPECIAL
+Asteroid Coating
+
+ASTEROID_COATING_OWNED_SPECIAL_DESC
+'''An outpost or colony on these asteroids will grand a structure bonus of +10% to all ships with Asteroid hulls in the empire.
+
+These asteroids are infected by transdimensional worms that eat tunnels through the rock, leaving behind a sticky slime that seems to automatically spread over the whole surface of the affected asteroid and prevents it from breaking appart.
+Applied to ships build with asteroid hulls the slime will naturally increase their durability. While scientists were unable to reproduce the substance or catch a living worm, they found out that just a small amount of fresh slime is enough to enhance whole fleets.'''
```
Comment (MatGB) [inline default/stringtables/en.txt]: built

**Comment by MatGB** [comment]
Wandering if it might be better to have it only apply in the same way weapon refinements are, as is all ships get the bonus immediately you capture the belt. Looks cool tho.

**Comment by Deepsloth** [comment]
It works like Reinforced Hull. They do get the bonus for max structure, but need to be "repaired" to actually get the bonus to current structure.

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -6373,6 +6373,24 @@ PHILOSOPHER_SPECIAL_DESC
...
+
+ASTEROID_COATING_OWNED_SPECIAL
+Asteroid Coating
+
+ASTEROID_COATING_OWNED_SPECIAL_DESC
+'''An outpost or colony on these asteroids will grand a structure bonus of +10% to all ships with Asteroid hulls in the empire.
+
+These asteroids are infected by transdimensional worms that eat tunnels through the rock, leaving behind a sticky slime that seems to automatically spread over the whole surface of the affected asteroid and prevents it from breaking appart.
+Applied to ships built with asteroid hulls the slime will naturally increase their durability. While scientists were unable to reproduce the substance or catch a living worm, they found out that just a small amount of fresh slime is enough to enhance whole fleets.'''
+
```
Comment (dbenage-cx) [inline default/stringtables/en.txt:6390]: If this is the same text, would it be better for translators to either replace with `[[ASTEROID_COATING_SPECIAL_DESC]]` or change the definition to use that description?

**Comment by dbenage-cx** [comment]
When an empire loses the source planet, the effect will no longer apply for such ships.
This seems inconsistent with a "coating" from a player perspective (unless there is some type of link that maintains the effect).
I'm not pushing for ship specials here, maybe the description can be reworded in some manner.  (May be needlessly complicating things for a minor issue)

**Comment by Deepsloth** [comment]
I did try to explain this by sneaking the word fresh into the description, but well... there are inconsistencies any way you look at it. The whole transdimensional worm story is quite silly i must admid, constructed for gameplay reasons.

Feel free to come up with something better. I won't mind scrapping the worm story.

File: default/scripting/specials/ASTEROID_COATING_OWNED.focs.txt [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    effectsgroups = [
+
+        // Gives a 10% structure bonus to all ships with asteroid hulls in the empire 
+        EffectsGroup
+            scope = And [
+                Ship
+                OwnedBy empire = Source.Owner
+                HasTag name = "ASTEROID_HULL"
+            ]
+            effects = SetMaxStructure value = Value + 0.1*abs(Value) accountinglabel = "ASTEROID_COATING_SPECIAL"
```
Comment (dbenage-cx) [inline default/scripting/specials/ASTEROID_COATING_OWNED.focs.txt]: Drop `abs`.  MaxStructure should be safely assumed as a positive value here (if not, other scripts may need revising).

File: default/stringtables/en.txt [hunk 5]
```diff
@@ -6373,6 +6373,24 @@ PHILOSOPHER_SPECIAL_DESC
 
 The lone remnant of a planet-sized space wandering species has found his final rest orbiting in this system like a normal planet. Having lived for thousands of years he lost the will and ability to travel and resorts to pondering about the mysteries of the universe he has seen and learned about. Eccentric and slightly demented he refuses to communicate with anything that isn't the size and shape of his kin, leaving scientists no choice but to ask questions via the planet they are living on. Understandably colonizing the Philosopher Planet will disturb him gravely and he will fall silent. Construction on his surface is difficult as is to be expected from a living being.'''
 
+ASTEROID_COATING_SPECIAL
+Asteroid Coating
+
+ASTEROID_COATING_SPECIAL_DESC
+'''An outpost or colony on these asteroids will grand a structure bonus of +10% to all ships with Asteroid hulls in the empire.
```
Comment (dbenage-cx) [inline default/stringtables/en.txt]: `grand` -> `grant`

File: default/stringtables/en.txt [hunk 6]
```diff
@@ -6373,6 +6373,24 @@ PHILOSOPHER_SPECIAL_DESC
...
+
+These asteroids are infected by transdimensional worms that eat tunnels through the rock, leaving behind a sticky slime that seems to automatically spread over the whole surface of the affected asteroid and prevents it from breaking appart.
```
Comment (dbenage-cx) [inline default/stringtables/en.txt]: `appart` -> `apart`

**Comment by Deepsloth** [comment]
@dbenage-cx : I went with your description in the forum. Thanks for all the suggestions.

**Comment by Vezzra** [comment]
I guess this will also require some AI adjustments, so we should open a respective AI issue when merging - @Morlic-fo?

**Comment by Morlic-fo** [comment]
Yes.

**Comment by dbenage-cx** [comment]
@MatGB bump?

**Comment by Vezzra** [comment]
Superseded by #3649

**Pull request closed without merging**